Making blood less frustrating

Kush1Push1 opened the pull request:
## About The Pull Request

Two changes.
```
1. If a splatter of blood isn't bloody enough, the character will not get bloody shoes.

2. Characters with light steps trait won't get bloody clothes if they step on a bloody tile without shoes.

3. added: Blood dries over time. Usually, 3 or 6 minutes. If a splatter is big enough, it can stay wet up to 15 minutes.
```

## Why It's Good For The Game

The entire crew won't have bloody shoes anymore.
Less time wasted looking for showers or sinks to wash their shoes after stepping on a near invisible blood decal.
Characters without shoes (or who don't want to get their clothes bloody after taking them off) will get more use from a near useless trait.

## Changelog
:cl:
qol: Blood splatters only make your shoes or feet bloody if the tile is bloody enough,
qol: Light steps trait prevents clothing from being covered in blood if you step on a bloody tile with your shoes off.
add: Blood dries over time. Usually, 3 or 6 minutes. If a splatter is big enough, it can stay wet up to 15 minutes.
/:cl:

(grammaticly impoved by @maxymax13)

Commit:
Initial

vpellen commented:
Thank you I'm massively autistic about keeping my feet clean
Is it possible to just have blood "dry out" over time, at which point it stops leaving prints altogether?

maxymax13 commented:
> Is it possible to just have blood "dry out" over time, at which point it stops leaving prints altogether?

that would be great

maxymax13 commented:
## About The Pull Request

Two changes.

    1. If a splatter of blood isn't bloody enough, the character will not get bloody shoes.

    2. Characters with light steps trait won't get bloody clothes if they step on a bloody tile without shoes.


( :) )
## Why It's Good For The Game

The entire crew won't have bloody shoes anymore.
Less time wasted looking for showers or sinks to wash their shoes after stepping on a near invisible blood decal.
Characters without shoes (or who don't want to get their clothes bloody after taking them off) will get more use from a near useless trait.
## Changelog

🆑
tweak: Blood splatters only make your shoes or feet bloody if the tile is bloody enough,
tweak: Light steps trait prevents clothing from being covered in blood if you step on a bloody tile with your shoes off.
/🆑

Farquaar commented:
This is good. When everybody has bloody shoes, no one has bloody shoes.
Maybe now, having bloody shoes will actually be uncommon enough to make it slightly suspicious.

Kush1Push1 commented:
@maxymax13 Thank you!

Fox-McCloud commented:
you could make it so that there's a timer created for blood---after a certain amount of time the timer pops and the blood dries, therefore preventing it from making shoes icky.

coiax commented on code/modules/detectivework/detective_work.dm:
```diff
@@ -81,6 +81,8 @@
 	return TRUE //we bloodied the floor
 
 /mob/living/carbon/human/add_blood_DNA(list/blood_dna, list/datum/disease/diseases)
+	if(HAS_TRAIT(src, TRAIT_LIGHT_STEP)) //the character is agile enough to don't mess their clothing and hands just from one blood splatter at floor
+		return TRUE
```
So this proc, `add_blood_DNA` is _only_ called by walking over a blood splatter, and no other times?

Or does this change give LIGHT_STEP trait holders the ability to not get bloody anymore?

Kush1Push1 replied:
Bloodysoles.dm is the only file that has references to `add_blood_DNA`, yeah.
But I still fear something could be overlocked, so I moved the code highter

Commit:
Drying

Adding mechanics to allow blood slowly dry.

Commit:
Fixing issue

(From the repo maint)

Kush1Push1 commented:
@coiax Could you review again?

Kush1Push1 commented:
Uh, right, I added things
Updating the description
Added:
added: Blood dryes over time. Usually, 3 or 6 minutes. If a splatter is big enough, it can stay wet up to 15 minutes.

Kush1Push1 commented:
I also tested how it works with virus. You still get infection without shoes (as intended) if have the trait.

Kush1Push1 commented:
Some screenshots
<details>

![dreamseeker_nP7K1ENX9U](https://user-images.githubusercontent.com/78963858/109998365-bd787100-7d43-11eb-96e4-8ec39f7fde75.png)
![dreamseeker_WIJC1gGncY](https://user-images.githubusercontent.com/78963858/109998381-c10bf800-7d43-11eb-8431-82fd2a3ef87f.png)
![dreamseeker_ll1dLd0gas](https://user-images.githubusercontent.com/78963858/109998489-da14a900-7d43-11eb-82e0-31843f96c95c.png)

</details>

Kush1Push1 commented:
Oof .-.'

MrMelbert commented:
just a warning, `tweak` tag in changelogs doesn't exist anymore, the tag is now QOL (and the tag for features is `add`
```
qol: Blood splatters only make your shoes or feet bloody if the tile is bloody enough,
qol: Light steps trait prevents clothing from being covered in blood if you step on a bloody tile with your shoes off.
add: Blood dryes over time. Usually, 3 or 6 minutes. If a splatter is big enough, it can stay wet up to 15 minutes.
```

Kush1Push1 commented:
> just a warning, `tweak` tag in changelogs doesn't exist anymore, the tag is now QOL (and the tag for features is `add`, not added)
> 
> ```
> qol: Blood splatters only make your shoes or feet bloody if the tile is bloody enough,
> qol: Light steps trait prevents clothing from being covered in blood if you step on a bloody tile with your shoes off.
> add: Blood dries over time. Usually, 3 or 6 minutes. If a splatter is big enough, it can stay wet up to 15 minutes.
> ```

Thank you!

optimumtact commented:
I'm not a huge fan of the use of timers because there can be a lot of blood decals on a bad round.

do we have an existing subsystem these could process on?

Commit:
Replacing timer with process()

Kush1Push1 commented:
> I'm not a huge fan of the use of timers because there can be a lot of blood decals on a bad round.
> 
> do we have an existing subsystem these could process on?

Replaced with process() on obj subsystem.
